drivers/storage/portworx/manifest/common.go

```diff
@@ -3,20 +3,54 @@ package manifest
 import (
 	"io/ioutil"
 	"net/http"
+	"net/url"
+	"strings"
 
-	yaml "gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	httpPrefix = "http://"
 )
 
 // Methods to override for testing
 var (
 	httpGet = http.Get
 )
 
-func getManifestFromURL(manifestURL string) ([]byte, error) {
-	resp, err := httpGet(manifestURL)
-	if err != nil {
-		return nil, err
+func getManifestFromURL(manifestURL string, proxy string) ([]byte, error) {
+	var resp *http.Response
+	var err error
+	if proxy == "" {
+		resp, err = httpGet(manifestURL)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Add the http prefix so we can parse the URL:
+		// Wrong format: url.Parse("127.0.0.1:3213")
+		// Correct format: url.Parse("http://127.0.0.1:3213")
+		if !strings.HasPrefix(strings.ToLower(proxy), httpPrefix) {
```

Comment:
Shouldn't we look for https proxy as well?

Reply:
good point, resolved in a new commit. pls take another look